### DISCUSSION TITLE
NATS account not found

### DISCUSSION THREAD

**Discussion by pcsegal** [nats-io/nats-server#5805] [Q&A]
Hi, I tried to create a NATS account just like I create any other account, by adding something similar to this to the "accounts" section in the config:

```
"account_name": {
    "jetstream": enabled,
    "users": [ {"user": "username", "pass": "password"} ]
}
```

However, when connecting as the user with the correct password and creating streams, I keep getting the error `nats: account not found`. Does anyone know why one would get that error?

Note: In this same NATS installation, if I remember correctly, it may be the case that there was an account with the same name that was removed from the config a long time ago and its files were removed from disk. Could this be causing the current issue?

Server version is 2.10.17, and it''s configured as a cluster with 5 pods (deployed on Kubernetes).

**Comment by derekcollison** [comment]
Would need quite a bit more information about your system..

Server version
System topology

**Reply by pcsegal** [reply]
Edited the question to add that the server version is 2.10.17 and the topology is a cluster with 5 pods (deployed on Kubernetes). I'm not sure how much additional information I should give, but this is a stripped down version of the config file used in each node:

```
# NATS Clients Port
port: 4222
# PID file shared with configuration reloader.
pid_file: "/var/run/nats/nats.pid"

http: 8222

server_name:$POD_NAME

jetstream {
    max_mem: 1Gi
    store_dir: /data
    domain: system-a
    max_file: 10Gi
}

lame_duck_grace_period: 10s
lame_duck_duration: 30s

cluster {
    port: 6222
    name: nats_cluster
    routes = [
        nats://nats-0.nats:6222,
        nats://nats-1.nats:6222,
        nats://nats-2.nats:6222,
        nats://nats-3.nats:6222,
        nats://nats-4.nats:6222,
    ]

    cluster_advertise: $CLUSTER_ADVERTISE
    no_advertise: true
    connect_retries: 120
}

accounts: {

    "$SYS": {
        "users": [
            {
                "pass": "password",
                "user": "sys"
            }
        ]
    },
    "account_a {
        "jetstream": "enabled",
        "users": [
            {
                "pass": "password",
                "user": "user_1"
            }
        ]
    },
    "account_b": {
        "jetstream": "enabled",
        "users": [
            {
                "pass": "password",
                "user": "user_2"
            }
        ]
    },
}
```

I need to see if I can reproduce the issue on a separate setup.

**Reply by derekcollison** [reply]
And if you use NATS cli with a context for either account_a or account_b you are getting jetstream not enabled?

What does `nats account info` show with the context?

**Reply by pcsegal** [reply]
When I do `nats account info`, it appears normal, as an empty JetStream account:

```
Connection Information:

               Client ID: 224
               Client IP: 10.244.5.238
                     RTT: 455.124µs
       Headers Supported: true
         Maximum Payload: 1.0 MiB
       Connected Cluster:cluster_name_here
           Connected URL: nats://nats:4222
       Connected Address: 10.244.4.192:4222
     Connected Server ID: NCJ73LOMJLZU6VND5T6EY5RYZFVJWAJVYCDR6CZVCN4AGTCDMPGAHSVF
   Connected Server Name: nats-0

JetStream Account Information:

Account Usage:

    Storage: 0 B
     Memory: 0 B
    Streams: 0 B
  Consumers: 0 B

Account Limits:

   Tier: Default

      Configuration Requirements:

         Stream Requires Max Bytes Set: false
          Consumer Maximum Ack Pending: Unlimited

      Stream Resource Usage Limits:

                    Memory: 0 B of Unlimited
         Memory Per Stream: Unlimited
                   Storage: 0 B of Unlimited
        Storage Per Stream: Unlimited
                   Streams: 0 of Unlimited
                 Consumers: 0 of Unlimited
```

When I run `stream ls`, I get `No Streams defined`.
So it all looks like the JetStream is enabled and working. It's only when I try to create a stream that I get the "account not found" error.

**Reply by pcsegal** [reply]
Note that this only happens with this newly added account; the other ones work normally as they did before, and allow adding streams.

**Reply by derekcollison** [reply]
Is it possible the new account is not loaded on all the servers?

How do you resolve accounts for your system?

**Reply by pcsegal** [reply]
I suspect not because I use a config file, then all pods of the StatefulSet are restarted with a rolling restart. I currently only configure the accounts in the server config file.

**Reply by pcsegal** [reply]
After further investigation I found out that the problem was indeed related to a node that hadn't correctly picked up the configuration. Thank you, closing this.
